client/lib/navigate/index.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+// Using page() for cross origin navigations would throw a `History.pushState` exception
+// about creating state object with a cross-origin URL.
+function isSameOrigin( path: string ): boolean {
```

Comment:
I think we can re-use `lib/url/is-external::isExternal()` for this?

Reply:
I don't like `isExternal()` very much. It feels too complicated for this purpose, does checks for some legacy routes that I believe are no longer necessary... I prefer the one-liner simplicity of the `isSameOrigin` here.